exercises/binary-search/example.scala

```diff
@@ -0,0 +1,23 @@
+import scala.annotation.tailrec
+
+object BinarySearch {
+  def search[T](seq: Seq[T], value: T)(implicit ord: T => Ordered[T]): Option[Int]
```

Comment:
I also prefer `Seq` over `Array`.
But does `x-common/binary-search` not even insist on `Array`?
(cf. the comment at the top in `canonical-data.json`).